src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/PackageItemViewModel.cs

```diff
@@ -490,18 +492,11 @@ public PackageLevel PackageLevel
             }
         }
 
-        public async Task<IReadOnlyCollection<VersionInfoContextInfo>> GetVersionsAsync()
-        {
-            var identity = new PackageIdentity(Id, Version);
-            var isTransitive = PackageLevel == PackageLevel.Transitive;
-            return await _searchService.GetPackageVersionsAsync(identity, Sources, IncludePrerelease, isTransitive, _cancellationTokenSource.Token);
-        }
-
         public async Task<IReadOnlyCollection<VersionInfoContextInfo>> GetVersionsAsync(IEnumerable<IProjectContextInfo> projects)
         {
-            var identity = new PackageIdentity(Id, Version);
             var isTransitive = PackageLevel == PackageLevel.Transitive;
-            return await _searchService.GetPackageVersionsAsync(identity, Sources, IncludePrerelease, isTransitive, projects, _cancellationTokenSource.Token);
+            await _packageModelVersions.PopulateDataAsync(Sources, IncludePrerelease, isTransitive, projects, _cancellationTokenSource.Token);
```

Comment:
I think `projects` can be `null` based on [this](https://github.com/NuGet/NuGet.Client/pull/6434/files#diff-118d6d1c3dcb2549a9436b8dafadcf915e0ae147a2ed54f3d82767252ee94701R688), but nullable isn't enabled in this file.

Also, https://github.com/NuGet/NuGet.Client/pull/6434/files#diff-fabe79ff0bb694bc6e954261e37d876f5a4dee275ec7c6c7bdc8a3ffc7540405R32 says `projects` must not be `null`.